### PR TITLE
32 migrate settings to orm and validator

### PR DESCRIPTION
**Pull request by Mathieu414**

### CODE REVIEW

File: utils/form_validation.php [hunk 1]
```diff
@@ -62,13 +70,35 @@ function get_field(string $key): Field
...
-            return $errors . "<br/><br/>";
+        return $result;
+    }
+
+    function set_success($success)
+    {
+        $this->success = $success;
+    }
+
+    function render_success()
```
Comment (Atmos4) [inline utils/form_validation.php:98]: Maybe consider merging this with `render_errors()` into a function like `render_meta()` or something

File: utils/form_validation.php [hunk 2]
```diff
@@ -489,4 +557,139 @@ function set_target_dir(string $directory)
         $this->target_dir = $directory;
         return $this;
     }
-}
\ No newline at end of file
+}
+
+class EmailField extends Field
```
Comment (Atmos4) [inline utils/form_validation.php:562]: This should imo extend StringField

Comment (Atmos4) [reply utils/form_validation.php:562]: Feel free to disregard if you think this is too much work, I can maybe look at this later and refactor

File: utils/form_validation.php [hunk 3]
```diff
@@ -489,4 +557,139 @@ function set_target_dir(string $directory)
...
+    }
+
+    function render(string $attrs = "")
+    {
+        $placeholder = $this->placeholder ?? $this->label;
+        return parent::render("type =\"email\" placeholder=\"$placeholder\"");
+    }
+}
+
+class PhoneField extends Field
```
Comment (Atmos4) [inline utils/form_validation.php:587]: Idem, maybe it should extend StringField 🤔

File: database/settings.api.php [hunk 4]
```diff
@@ -1,72 +1,10 @@
...
 
-function get_user_data()
-{
-    return fetch("SELECT * FROM licencies WHERE id = ? LIMIT 1;", $_SESSION['user_id'])[0];
-}
-
 function get_user_login($id)
 {
     return fetch("SELECT login FROM licencies WHERE id=? LIMIT 1;", $id)[0];
 }
```
Comment (Atmos4) [inline database/settings.api.php:6]: Look into removing that file maybe? The content is deprecated

File: pages/seed_db.php [hunk 5]
```diff
@@ -45,7 +45,7 @@ function map_gender(string $g): Gender
     $newUser->licence = $user['num_lic'];
     $newUser->nose_email = $user['email'];
     $newUser->real_email = $user['realmail'];
-    $newUser->phone = $user['telport'];
+    $newUser->phone = preg_replace("/[^0-9]/", "", $user['telport']);
```
Comment (Atmos4) [inline pages/seed_db.php:48]: phones can have a + in the number. Maybe we need to change the model instead 🤔

File: utils/form_validation.php [hunk 6]
```diff
@@ -44,10 +47,15 @@ function value(string $key)
 
     function valid(string $key = null)
     {
+        if (isset($_POST['action'])) {
+            $action_check = ($_POST['action'] != $this->action);
```
Comment (Atmos4) [inline utils/form_validation.php:51]: Imo this is not needed, this case is already covered in the constructor

File: utils/form_validation.php [hunk 7]
```diff
@@ -264,7 +332,7 @@ class DateField extends Field
 {
     function check(string $msg = null)
     {
-        if (!preg_match("/^[\d-]*$/", $this->value)) {
+        if (!preg_match("/^[\d-]*$/", $this->value ?? "")) {
```
Comment (Atmos4) [inline utils/form_validation.php:335]: Look into turning this generic line into a private method inside `Field`, like this:
```php
private function test($preg){
  return preg_match($preg, $this->value ?? "");
}
```

File: pages/seed_db.php (hunk 5 above)
Comment (Mathieu414) [reply pages/seed_db.php:48]: Yes that's true, especially if we have a foreign user

**Review by Atmos4 — APPROVED** [review]

**Review by Atmos4 — DISMISSED** [review]

**Review by Atmos4 — APPROVED** [review]

**Pull request merged**